### ISSUE TITLE
Make `v0.2.0` release

### ISSUE THREAD

**Issue by gpauloski** [proxystore/benchmarks#98]
Now that ProxyStore is on `v0.6.2`, it'd been good to make a new release of the benchmarks with all the dependencies updated to their latest versions.

- [ ] [Low priority] Support Python 3.12.
  - Blocked on `globus_compute_endpoint` restricting PyZMQ to older versions that don't support Python 3.12. See: https://github.com/funcx-faas/funcX/blob/b17d4683550c88ad1a97080cd929b0a7d1960adb/compute_endpoint/setup.py#L33
- [x] Update Globus Compute to newest version
- [x] Update ProxyStore to `v0.6.4` (requires new ProxyStore release)
- [x] Update code to use extensions package
- [x] Update Colmena (requires new Colmena release)
- [x] Issue #119 
- [x] Finish new benchmarks (streaming, workflows, memory, etc.)
